pkg/prometheus/promcfg.go

```diff
@@ -2142,6 +2163,46 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 	}
 }
 
+func generateRoleSelectorConfig(k8sSDConfig yaml.MapSlice, roles []string, selector *metav1.LabelSelector, cg *ConfigGenerator) yaml.MapSlice {
+	selectors := []yaml.MapSlice{}
+	for _, role := range roles {
+		yml := yaml.MapSlice{
+			{
+				Key:   "role",
+				Value: role,
+			},
+		}
+
+		labelValues := []string{}
+		if len(selector.MatchLabels) > 0 {
+			labelValues = append(labelValues, labels.FormatLabels(selector.MatchLabels))
+		}
+
+		if len(selector.MatchExpressions) > 0 {
+			expressions := []string{}
+
+			for _, exp := range selector.MatchExpressions {
+				// Needs to lower because the selection expects lowercase.
+				// exp.Operator is CamalCase.
+				requirement, err := labels.NewRequirement(exp.Key, selection.Operator(strings.ToLower(string(exp.Operator))), exp.Values)
+				if err != nil {
+					cg.logger.Error("failed to create label requirement", "err", err)
```

Comment:
Should we check earlier (in resource selector) that the input is correct?

Reply:
I don't think so since it's a LabelSelector and K8S API Will ensure it for us, right?

Reply:
Yeah, I don't have a concrete example where validation markers would fail us but it might be a good idea to prevent the potential issue though?